### PR TITLE
fix(dna): consistent allow_outside_chromosome defaults

### LINKED ISSUE #97: allow_outside_chromosome is default=True in python but actually defaults as False in the C layer
https://github.com/deepgenomics/GenomeKit/blob/8e8fca0237d647561637ef44fd7dc89b4370ee4a/genome_kit/genome_dna.py#L30

vs

https://github.com/deepgenomics/GenomeKit/blob/8e8fca0237d647561637ef44fd7dc89b4370ee4a/src/py_genome_dna.cpp#L63

Also VariantGenome.dna doesn't support it

### PR DESCRIPTION
**Pull request by s22chan**
fixes #97

### CODE REVIEW

**Comment by ovesh** [comment]
I guess we need to remove 3.8 tests before merging this

File: src/genome_dna.h [hunk 1]
```diff
@@ -20,7 +20,7 @@ class genome_dna {
 	INLINE bool is_open()     const { return _fmap.is_open(); }
 	INLINE void ensure_open() const { if (!is_open()) open_on_demand(); }  // Fast check if file / indices already loaded
 
-	dnastr operator()(const interval_t& c, bool allow_outside_chromosome=false) const;
```
[the remark this answers is not in the document]
Comment (s22chan) [reply src/genome_dna.h:23]: what do you mean by the wrong default?  you mean a C level test? we don't rely on the C level API for testing

**Comment by s22chan** [comment]
> I guess we need to remove 3.8 tests before merging this

I think there's actually a test behavioural change but I won't be looking at this anytime soon

**Comment by ovesh** [comment]
> > I guess we need to remove 3.8 tests before merging this
> 
> I think there's actually a test behavioural change but I won't be looking at this anytime soon

I'm just assuming the 3.8 tests are failing due to the removal of support for py38 in conda-forge (some change in meta.yaml)

File: src/genome_dna.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (ovesh) [reply src/genome_dna.h:23]: I thought this fix has an actual effect.

**Review by ovesh — APPROVED** [review]

**Pull request merged**